core/commands/destroy_command.rb

```diff
@@ -166,4 +191,45 @@ def execute
     end
     SUCCESS_RESULT
   end
+
+  private
+
+  def remember_aws_instance_id(configuration, node)
+    return if configuration.provider != 'aws'
+
+    @aws_instance_ids = {}
+    if node.empty?
```

Comment:
If node name is specified, create a temporary array and use it to crate the hash for both cases.